functionalDB/chapter.md

```diff
@@ -0,0 +1,1138 @@
+</li>
+</ul></td>
+  </tr>
+  <tr>
+    <td>VEAT index</td>
+    <td>VAET index</td>
+  </tr>
+  <tr>
+    <td><ul>
+<li>
+<span style="background-color:pink">Rome</span> ⇒ {<span style="background-color:lightblue">JC</span> ⇒ {<span style="background-color:lightgreen">lives-in</span>}}<br/>
+{<span style="background-color:lightblue">B</span> ⇒ {<span style="background-color:lightgreen">lives-in</span>}}
+</li>
+<li>
+<span style="background-color:pink">Egypt</span> ⇒ {<span style="background-color:lightblue">Cleo</span> ⇒ {<span style="background-color:lightgreen">lives-in</span>}}
+</li>
+<li>
+<span style="background-color:pink">Tiber</span> ⇒ {<span style="background-color:lightblue">Rome</span> ⇒ {<span style="background-color:lightgreen">river</span>}}
+</li>
+<li>
+<span style="background-color:pink">Nile</span> ⇒ {<span style="background-color:lightblue">Egypt</span> ⇒ {<span style="background-color:lightgreen">river</span>}}
+</li></ul></td>
+<td><ul>
+<li>
+<span style="background-color:pink">Rome</span> ⇒ {<span style="background-color:lightgreen">lives-in</span> ⇒ {<span style="background-color:lightblue">JC, B</span>}}
+</li>
+<li>
+<span style="background-color:pink">Egypt</span> ⇒ {<span style="background-color:lightgreen">lives-in</span> ⇒ {<span style="background-color:lightblue">Cleo</span>}}</li>
+<li>
+<span style="background-color:pink">Tiber</span> ⇒ {<span style="background-color:lightgreen">river</span> ⇒ {<span style="background-color:lightblue">Rome</span>}}
+</li>
+<li>
+<span style="background-color:pink">Nile</span> ⇒ {<span style="background-color:lightgreen">river</span> ⇒ {<span style="background-color:lightblue">Egypt</span>}}
+</li></ul></td>
+  </tr>
+</table>
+Table 2
+
+### Database
+
+We now have all the components we need to construct our database! Initializing our database means:
+
+* creating an initial empty layer with no data 
+* creating a set of empty indexes
+* settings its top-id and curr-time to be 0
+````clojure
+(defn ref? [attr] (= :db/ref (:type (meta attr))))
+
+(defn always[& more] true)
+
+(defn make-db []
+   (atom 
+       (Database. [(Layer.
+                   (fdb.storage.InMemory.) ; storage
+                   (make-index #(vector %3 %2 %1) #(vector %3 %2 %1) #(ref? %)); VAET                     
+                   (make-index #(vector %2 %3 %1) #(vector %3 %1 %2) always); AVET                        
+                   (make-index #(vector %3 %1 %2) #(vector %2 %3 %1) always); VEAT                       
+                   (make-index #(vector %1 %2 %3) #(vector %1 %2 %3) always); EAVT
+                  )] 0 0)))
+````
+There is one snag, though -- all collections in Clojure are immutable. Since write operations are pretty critical in a database, we call **atom** on our structure first. This is one of Clojure’s reference types, and it provides atomic writes to the element it wraps. 

```

Comment:
Actually we don't call Atom on the structure, but define it to be an Atom, which is one of Clojure's reference types. I'll update the text.